apps/namadillo/src/lib/query.ts

```diff
@@ -91,22 +91,30 @@ export const buildTxArray = async <T>(
 
   if (!publicKey) {
     const revealPkTx = await tx.buildRevealPk(wrapperTxProps, account.address);
-    txArray.push({ type: revealPublicKeyType, encodedTx: revealPkTx });
+    txArray.push({ type: revealPublicKeyType, tx: revealPkTx.tx });
```

Comment:
Just a question, but is there a reason for not including the reveal PK TX in the batch?

Reply:
Yes! I separated reveal PK so that the batch contains one Tx type, so it has its own approval for now to make it clear that it is a Tx that needs to be approved and sent. Maybe we would want to include it, but for displaying details and such, it's much easier to think about if a batch is of one Tx type

Reply:
I think there is no technical reason that would prevent us from adding it into the batch, but from a UX point of view, the interface always sends a batch of a single tx type at a time. If we decide we want to do it, it will be trivial to add it in. I will ask!